### ISSUE TITLE
Weierstrass polynomial

### ISSUE THREAD

**Issue by thofma** [oscar-system/Oscar.jl#2035]
There was a request for Weierstrass polynomials on slack. It seems this could be addressed if we could somehow wrap the following Singular functionality: https://www.singular.uni-kl.de/Manual/4-0-3/sing_2001.htm.

**Comment by HereAround** [comment]
It seems that I missed that post, so I cannot fully gauge the background.

Still, Weierstrass polynomials (in the context of elliptic fibrations and tailor-made for my research interests, i.e. F-theory) are available in https://github.com/Julia-meets-String-Theory/FTheoryTools.jl. If things go according to plan, we shall begin to gauge an integration of this software into OSCAR in a couple of weeks.

**Comment by thofma** [comment]
I think this is not about Weierstrass models of elliptic curves, but [https://www.google.com/search?q=weierstrass+polynomial](https://www.google.com/search?q=weierstrass+polynomial).

**Comment by HereAround** [comment]
Indeed, that is separate from elliptic curves as Weierstrass models. (In my universe, we often speak of the Weierstrass polynomial in this context. But I guess this one could attribute to being lazy). Sorry for sidetracking the discussion. Ignore my post above.

**Comment by afkafkafk13** [comment]
indeed he is asking precisely for the functionality of weierstr.lib. Wrapping it would be one option, but this is actually power series functionality (up to given precision) encoded in Singular using the localization of the polynimial ring at the origin and cutting off at a given power. So we might have better tools and data structures for these (not very elaborate) methods.

**Comment by wdecker** [comment]
I will wrap this as soon as I find the time to do it.

**Comment by fingolfin** [comment]
@wdecker  will discuss with @afkafkafk13  how to proceed

**Comment by fingolfin** [comment]
Just brought this up during triage, as this was simply forgotten. @wdecker will have a look, perhaps @afkafkafk13 is also still interested.

**Comment by afkafkafk13** [comment]
We'll discuss this in the AlgGeo-Meeting later on today.

**Comment by afkafkafk13** [comment]
For the time being, this will just be wrapped from Singular -- @wdecker offered in the triage meeting to do this. In the long run, this should move over to Oscar, as improving the precision of the power series amounts to complete recomputation, if wrapped to Oscar.
